plugins/Ecommerce/Columns/Revenue.php

```diff
@@ -27,6 +33,55 @@ public function getDbDiscriminator()
         return new Discriminator($this->dbTableName, 'idgoal', GoalManager::IDGOAL_ORDER);
     }
 
+
+    public function configureSegments(SegmentsList $segmentsList, DimensionSegmentFactory $dimensionSegmentFactory)
+    {
+        //new Segment revenue on order
+        $segment = new Segment();
+        $segment->setCategory($this->category);
+        $segment->setName(Piwik::translate('Ecommerce_OrderRevenue'));
+        $segment->setSegment('revenueOrder');
+        $segment->setSqlSegment('log_conversion.idvisit');
+        $segment->setSqlFilter(function ($valueToMatch, $sqlField, $matchType) {
+            return $this->getRevenueQuery($valueToMatch, $sqlField , $matchType, 0);
+        });
+        $segmentsList->addSegment($dimensionSegmentFactory->createSegment($segment));
+
+        //new Segment revenue left in cart
+        $segment = new Segment();
+        $segment->setCategory($this->category);
+        $segment->setName(Piwik::translate('Ecommerce_RevenueLeftInCart'));
+        $segment->setSegment('revenueAbandonedCart');
+        $segment->setSqlSegment('log_conversion.idvisit');
+        $segment->setSqlFilter(function ($valueToMatch, $sqlField , $matchType) {
+           return $this->getRevenueQuery($valueToMatch, $sqlField , $matchType,-1);
+        });
+        $segmentsList->addSegment($dimensionSegmentFactory->createSegment($segment));
+
+    }
+
+    /**
+     * revenue sql filter
+     * @param $valueToMatch
+     * @param $sqlField
+     * @param $matchType
+     * @param false $cart
+     * @return array
+     */
+    private function getRevenueQuery($valueToMatch, $sqlField , $matchType, $idGoal)
+    {
+        if ($matchType === SegmentExpression::MATCH_EQUAL) {
+            $matchType = "=";
+        }
+        $table = Common::prefixTable($this->dbTableName);
+        $sql = " SELECT {$sqlField} from {$table} WHERE (idgoal = {$idGoal} and log_conversion.revenue {$matchType} ?) ";
```

Comment:
Did you really test that one? When adding such segment queries you should always check the queries that are generated.

This query is actually not doing what you might expect, at least not if matomo is using a table prefix. Assuming someone uses the table prefix `matomo_`, this query would be like `SELECT log_conversion.idvisit FROM matomo_log_conversion WHERE (idgoal = 0 and log_conversion.revenue <= 0)`

Looking e.g. at the query that is generated to fetch the visits log this results in:
```sql
SELECT /* idSegments = [13] */ log_visit.*
FROM matomo_log_visit AS log_visit
LEFT JOIN matomo_log_conversion AS log_conversion ON log_conversion.idvisit = log_visit.idvisit
WHERE 
  (log_visit.idsite in (?) AND log_visit.visit_last_action_time >= ? AND log_visit.visit_last_action_time <= ?) 
AND (
    (log_conversion.idvisit IN (
       SELECT log_conversion.idvisit
         FROM matomo_log_conversion
         WHERE (idgoal = 0 AND log_conversion.revenue >= ?)
)))
GROUP BY log_visit.idvisit
ORDER BY log_visit.idsite DESC, log_visit.visit_last_action_time DESC
LIMIT 0, 27
```
You might notice that the subquery now actually references some fields from the outer query.